api/v1beta1/barbican_webhook.go

```diff
@@ -202,6 +195,15 @@ func (r *BarbicanSpec) ValidateUpdate(old BarbicanSpec, basePath *field.Path) fi
 		basePath.Child("barbicanAPI").Child("override").Child("service"),
 		r.BarbicanAPI.Override.Service)...)
 
+	// pkcs11 verifications
+	if slices.Contains(r.EnabledSecretStores, SecretStorePKCS11) {
+		if r.PKCS11 == nil {
+			allErrs = append(allErrs, field.Required(basePath.Child("PKCS11"),
+				"PKCS11 specification is missing, PKCS11 is required when pkcs11 is an enabled SecretStore"),
+			)
+		}
+	}
```

Comment:
This is the same code as line 141 above, which makes me think it may be worth moving into a common place.

Reply:
Done